### ISSUE TITLE
Name like "J. P. Gownder" incorrectly parsed

### ISSUE THREAD

**Issue by robertandrews** [ADCI/full-name-parser#5]
In the instance of "J. P. Gownder" (with space between "J." and "P."), "J. P." is actually a first name.

However, full-name-parser interprets the following:
First = "J."
Middle: "P."

Would parsing "J. P.", or correcting to "J.P.", as first name cause conflict with other rules?

**Comment by Deyter** [comment]
@robertandrews
Are you sure that it is the first name? What is the full name of Mr. Gownder?
And there's no rule that you can make initials only from the first name. For that reason, if it is a compound first name you can write it with dots but without spaces between initials.
In future versions, we planned to add functionally to parse compound first names like `John Paul`. But in this case, we need to know the full version of name, not initials.

**Comment by robertandrews** [comment]
> Are you sure that it is the first name? What is the full name of Mr. Gownder?

For all I know, he was born "John" (first) "Paul" (middle) "Gownder" (last) - but "J. P. Gownder" is absolutely how he refers to himself across the internet. There is some inconsistency across sites (Eg. "J.P." (without space) vs "J. P." (with space) - but the latter is the one he uses himself.

There's no other name known.

**Comment by anton87gp** [comment]
@robertandrews 
As you mentioned, Paul is middle name, so, full-name-parser parses it correctly :)

If to google "J. P. Gownder" about a half of results is "J.P. Gownder" and another half is "J. P. Gownder" so the name is mentioned in inconsistent ways, yeah.
Naming should conform some conventions anyway to be parsed correctly. Even if it's how he refers to himself across the internet this doesn't mean that this name conforms conventions.
As it was mentioned, if you'll remove space then "J.P." will be parsed as first name - hope this will work for you in this particular case.
Basically there is nothing wrong in how full-name-parser parses this particular name.

**Comment by robertandrews** [comment]
Alright. I willl experiment with trim.
